frontend/src/Explorer/View/Blocks.purs

```diff
@@ -86,56 +92,34 @@ emptyBlocksView message =
         [ P.dangerouslySetInnerHTML message ]
         []
 
-latestBlocks' :: State -> CBlockEntries
-latestBlocks' state =
-    case state ^. latestBlocks of
+failureView :: Language -> P.Html Action
+failureView lang =
+    P.div
+        []
+        [ P.p
+            [ P.className CSS.blocksFailed ]
+            [ P.text $ translate (I18nL.block <<< I18nL.blNotFound) lang ]
+        , P.link (toUrl Dashboard)
+            [ P.className "btn-back" ]
+            [ P.text $ translate (I18nL.common <<< I18nL.cBack2Dashboard) lang ]
+        ]
+
+
+
+unwrapLatestBlocks :: RemoteData Error CBlockEntries -> CBlockEntries
+unwrapLatestBlocks blocks =
+    case blocks of
             Success blocks' -> blocks'
             _ -> []
 
 currentBlocks :: State -> CBlockEntries
 currentBlocks state =
-    if expanded
-    then slice minBlockIndex (minBlockIndex + maxBlockRows) blocks
-    else slice 0 minBlockRows blocks
+    slice minBlockIndex (minBlockIndex + maxBlockRows) blocks
     where
-        blocks = latestBlocks' state
-        expanded = state ^. dashboardBlocksExpanded
-        currentBlockPage = state ^. (dashboardViewState <<< dashboardBlockPagination)
+        blocks = unwrapLatestBlocks $ state ^. currentBlocksResult
```

Comment:
@sectore All good. I misread `currentBlocksResult` and thought it was `currentBlockPage`.